### ISSUE TITLE
Add a "Rude flag" link next to the "Spam flag" link

### ISSUE THREAD

**Issue by user12986714** [Charcoal-SE/metasmoke#722]
![image](https://user-images.githubusercontent.com/65436504/82122041-1a784b80-975f-11ea-9efd-4fcdb8637187.png)
At the very end of a MS report, there is a link of which one can cast a spam flag. Would we consider adding a "Rude flag" link in addition to that, as it is suggested not to use spam flags for rude/abusive content (and a user may not want to firstly reply k in chat and then go to the site to flag)?

**Comment by ArtOfCode-** [comment]
> as it is suggested not to use spam flags for rude/abusive content

Only by those who are unnecessarily pedantic about the types of two red flags which are functionally equivalent in almost every way.

Spam vs. abusive makes no difference; you can flag spam as abuse or abuse as spam, in theory, and it should get marked helpful. If you're concerned that it won't, you can always use the link to the post on-site to cast your preferred type of flag, at the cost of a few extra clicks and ten seconds.

**Comment by user12986714** [comment]
The main problem here is that users need to spend a little more time and effort, and therefore creates a tiny friction for users to use appropriate flag. The difference is minor for a few times, but in the long run the accumulated effect may not be negligible, and hence that tiny friction may shift users into trending to flag posts as spam, rather than using an appropriate flag.

The use of appropriate flags may be important, as spam and rude/abusive contents are of very different nature. On a lower level, they are handled by the system in the same way. However, the information carried in those flags is not meant to only be used by the lower level system (SpamRam, etc.), but also to be used by Stack Exchange employees, moderators and other people involved in the moderation of the site; and on the higher level, they should be handled differently due to their distinct nature. Flagging contents indiscriminately as spam causes friction for those people of whom is also the intended audience of the information carried by the flags, and again, this may be minor for a few times, but in the long run, the effect may not be negligible.

**Comment by Undo1** [comment]
This has been discussed many, many times. Each time, we find that there is no or minimal benefit to implementing automatic R/A flags. In the many times I’ve seen this debate, there has never been shown a concrete case of harm arising from the use of only slam flags - that is, except the many words spent debating the issue itself. 

> the information carried in those flags is [...] to be used by Stack Exchange employees, moderators and other people involved in the moderation of the site

 - We’ve been told by CMs not to worry about the difference. 
 - Unfortunately, I don’t think the company is going to be doing anything around automated moderation tools soon, and certainly nothing that would care about spam v. R/A.
 - I’m an SO moderator, and have never seen a legitimate use case for caring about spam v. R/A. 
 - The people involved in the moderation of the site is fairly well represented by... us. I’m not aware of any friction caused to us by not respecting the split, other than the time spent debating it. 

In short, we’ve already been doing this for the long run. The effect - other than this very debate - *is* negligible.

**Comment by user12986714** [comment]
Closed as not being a valid issue for reasons suggested by ArtOfCode- and Undo1.

**Comment by gparyani** [comment]
I think this should be reconsidered. I've read through @ArtOfCode- and @Undo1's arguments against this request, and the main crux of it seems to be that the two red flag types are treated pretty much equivalently.

While that is true for the most part (especially in counting toward the 6-flag threshold), there are a couple of cases where the system *does* differentiate between those two flag types:

- Deleted answers with helpful spam flags count toward [automatic protection](https://meta.stackexchange.com/a/52765/377214) of the question. Answers that have helpful rude/abusive flags, but no helpful spam flags, [won't count](https://meta.stackexchange.com/a/58035/377214). This means that if a rude/abusive answer gets deleted and at least one user has flagged it as *spam* in particular (e.g. through the MS flagging system), it may result in the question being incorrectly protected when it's not supposed to. (This is more common than you may think: so long as there is at least one past answer from a new user deleted for any reason, even merely being NAA, a *single* answer flagged as spam from a new user will trigger auto-protection.)
- Posts with at least one helpful rude/abusive flag are excluded from being selected as review audits in the Triage and Low Quality Posts queues, to prevent NSFW posts from appearing as review audits. This means that, say, if an offensive post gets autoflagged (as spam) and not one other user flags it as rude/abusive, it can result in an NSFW review audit for others.

In summary, it's not true that there is no legitimate use case for caring about the flag types, or that comparing between the flag types is unnecessarily pedantic. While it would be difficult to implement R/A flags as actual autoflags, I hope that the request in the main issue post, to add a button to R/A flag, can be reconsidered.

**Comment by ArtOfCode-** [comment]
In the small number of cases it actually matters, you can just use the link to the post from the same page, and flag it there on site.

**Comment by ArtOfCode-** [comment]
It's not quite that easy. It's probably not too difficult, but having to open the post is not a big ask, so it's not really worth it.

**Comment by user12986714** [comment]
@ArtOfCode- I actually have a 19++ 10-- implementation locally. If anyone is willing to review it, I am happy to send a PR.

**Comment by makyen** [comment]
note: Also posted as [comments in chat](https://chat.stackexchange.com/transcript/message/57617825#57617825):

It seems there could be some benefit. At a minimum, some users will feel better about manually raising the "right" flag. I think the main objection to R/A flagging has been with respect to autoflagging and developing logic to automatically choose the "right" flag for the post, which this isn't. Giving humans the option to raise a rude/abusive flag seems reasonable, or at least not unreasonable, so doesn't feel like something that should be actively blocked from being developed.

At that point, it becomes a question of where to allocate developer time, but that's always been something open for people to choose to spend their time on things which interest them (or where needed for high priority items). If someone already has a solution available to add the feature, or even if someone just wants to put the effort in, I don't see why we shouldn't support a PR. Time to review the PR can be allocated as reviewers see fit, as it's not something which is a high priority.

Obviously, before it can be tested or deployed, there's the gating issue that work needs to be done to update dependencies in order for us to be able to build MS, due to the [mimemagic issue](https://github.com/Charcoal-SE/metasmoke/issues/849) which currently prevents both CI and actual builds.

**Comment by ArtOfCode-** [comment]
@user12986714 see above - happy to accept a PR
